spring-pulsar-spring-boot-autoconfigure/src/main/java/org/springframework/pulsar/autoconfigure/PulsarReactiveAutoConfiguration.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.autoconfigure;
+
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.reactive.client.adapter.AdaptedReactivePulsarClientFactory;
+import org.apache.pulsar.reactive.client.adapter.ProducerCacheProvider;
+import org.apache.pulsar.reactive.client.api.ReactiveMessageSenderCache;
+import org.apache.pulsar.reactive.client.api.ReactivePulsarClient;
+import org.apache.pulsar.reactive.client.producercache.CaffeineProducerCacheProvider;
+
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.pulsar.core.reactive.DefaultReactivePulsarSenderFactory;
+import org.springframework.pulsar.core.reactive.ReactivePulsarSenderFactory;
+import org.springframework.pulsar.core.reactive.ReactivePulsarSenderTemplate;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+
+/**
+ * {@link EnableAutoConfiguration Auto-configuration} for Apache Pulsar.
+ *
+ * @author Chris Bono
+ * @author Christophe Bornet
+ */
+@AutoConfiguration(after = PulsarAutoConfiguration.class)
+@ConditionalOnClass(ReactivePulsarSenderTemplate.class)
```

Comment:
We actually want to make it conditional on the reactive client. If/when we split out the core spring-pulsar module we could base the condition on the RPT. Currently it will always be available since it is in `spring-pulsar`. 

I believe this is also why the `compileOnly` thing had to be done. I changed this locally and was able to put the deps. back as `api`.

Reply:
Shouldn't it be both ReactivePulsarSenderTemplate and ReactivePulsarClient ?
If we have the reactive client but not spring-pulsar on the CP, this will fail...

Reply:
And with the comment https://github.com/spring-projects-experimental/spring-pulsar/pull/175#discussion_r1005889990, spring-pulsar will pull the reactive client so it's only spring-pulsar's presence we should test

Reply:
> Shouldn't it be both ReactivePulsarSenderTemplate and ReactivePulsarClient ?
Yes, this would be safer to do. I can add this in polish commit during merge though.